orttraining/orttraining/training_api/module.cc

```diff
@@ -185,6 +299,10 @@ Module::Module(const std::string& train_model_path_or_bytes,
     }
     eval_input_names_ = eval_user_input_names;
     eval_input_names_.insert(eval_input_names_.end(), eval_param_input_names.begin(), eval_param_input_names.end());
+
+    // Keep a copy of the eval model to be able to later export the model for inferencing.
```

Comment:
can we delay loading it? imagine the memory on device is small, we should delay load the copy once it is really needed.

Reply:
I think this can get a little tricky if the user has stored the eval file in a temporary dir on device (which can be cleared after `Module` creation). But maybe we can document it clearly so users make sure such a file exists when calling `ExportModelForInference`

Alternatively the api can take in an eval model file instead of it being stored as a variable in Module.

Reply:
Moved the load to happen during function invocation. Delayed load requires that the eval model still be at the location specified in the constructor. Added a note for that.